File: src/main/java/failchat/twitch/TwitchHighlightHandler.java
package failchat.twitch;

import failchat.core.MessageHandler;
import org.apache.commons.lang.StringUtils;

public class TwitchHighlightHandler implements MessageHandler<TwitchMessage> {
    private String appeal;

    public TwitchHighlightHandler(String channel) {
        this.appeal = '@' + channel;
    }

    @Override
    public void handleMessage(TwitchMessage message) {
        if (StringUtils.containsIgnoreCase(message.getText(), appeal)) {
            message.setHighlighted(true);
        }
    }
}

File: src/main/kotlin/failchat/twitch/TwitchHighlightHandler.kt
package failchat.twitch

import failchat.chat.MessageHandler

class TwitchHighlightHandler(channel: String) : MessageHandler<TwitchMessage> {

    private val appeal = "@" + channel

    override fun handleMessage(message: TwitchMessage) {
        if (message.text.contains(appeal, ignoreCase = true)) {
            message.highlighted = true
        }
    }
}
